gateway/src/management_switch/location_map.rs

```diff
@@ -224,9 +281,37 @@ impl ValidatedLocationConfig {
         // collection of reasons the config is invalid (if any)
         let mut reasons = Vec::new();
 
-        let names = vec_to_hashset(config.names, &mut reasons, || {
-            String::from("location names contain at least one duplicate entry")
-        });
+        let description = {
+            let n = config.description.len();
+            let description =
+                config.description.into_iter().collect::<IdOrdMap<_>>();
+
+            // Validate that we have no duplicate location names.
+            if n != description.len() {
+                reasons.push(String::from(
+                    "location descriptions contain at least one duplicate name",
+                ));
+            }
+
+            // Validate that we have no duplicate `local_sled` values.
+            let local_sled = description
+                .iter()
+                .map(|d| d.local_sled)
+                .collect::<HashSet<_>>();
+            if n != local_sled.len() {
+                reasons.push(String::from(
+                    "location descriptions contain at least one \
+                     duplicate local_sled",
```

Comment:
nit, take it or leave it: since `reasons` is a `Vec<String>` anyway, we may as well format the duplicated local sled here?